docs/JSON-RPC.md

```diff
@@ -1,14 +1,14 @@
 
-# Jamulus JSON-RPC Documentation
+# Jamulus JSON-RPC Server Documentation
 
 <!--
 This file is automatically generated from the source code.
 Do not edit this file manually.
 See `tools/generate_json_rpc_docs.py` for details.
 -->
```

Comment:
Given the above, why is it added to git?

Reply:
I want the finished page to be found at a fixed location on the internet.

Reply:
Maybe it should be handled under jamulus/jamuluswebsite?  I don't think we've anything else quite like this...

@gilgongo ?

Reply:
I think it was considered as "developer only" but if people think JSON RPC is ok to become a bit more official, we can surely add it to the website

Reply:
I'm not sure if there's a convention for this. If the majority of those who would use JSON RPC would naturally go to the repo to find the docs on it (as they would for compiling, or licensing, say), then that's an argument for keeping it in the repo. Otherwise it would make sense to have as much info as possible in one place (the website), I think?

Reply:
I don't think most of them would go to the repo by default. But I'd say they can be directed to the repo - but that isn't needed. 

However - this file is automatically generated and updated if the code changes. So that's something to consider.

Reply:
> However - this file is automatically generated and updated if the code changes. So that's something to consider.

And that's what I don't like - the generated files shouldn't be under git but deployed some other way as part of the build.  The Jamulus executable is generated as part of the build and people expect to come to the Github repo to get hold of it (some people who know about Github and the build/release process it has).  I'd like the same process to trigger deployment of these files, which can be referenced from a fixed place that _is_ held in git.

Reply:
If you're saying it's dodgy to rely on someone running this script to update this output file when code changes, I agree.

Reply:
If it's getting to Github, then it's going through the PR process.

That means it goes through the deployment process.

Which means the generated files don't need to be committed to git, just published, like the Jamulus executable gets published when it's built.

Reply:
There's inherent duplication now, which caused me some confusion, but I'm not sure what the change would look like.

Reply:
> We have developer docs at a fixed address that we can link to. What's wrong with that?

Yes - seems a bit too cosmetic to move the API docs to the website if the "journey" for anyone interested in using it is that they browse the website, see there's `--jsonrpcport` and then follow the link. I don't know if there's some SEO advantage in having it under jamulus.io, but that's likely to be very small.

The question of the update script and how that gets run is a separate matter though.

Reply:
> There's inherent duplication now, 

Is there?

Reply:
I was thinking of Github pages.  When the build runs, you can configure the workflow run to publish the file (see [here](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)).  We could publish under `http://jamulussoftware.github.io/jamulus/`, to differentiate from the "official" `https://jamulus.io/` site, using GitHub workflows to render the file(s).  That could be linked from the main GitHub README.md, if desired...

Reply:
I think that's confusing. It should in the best case open a PR with the changes on the website.

Reply:
> > There's inherent duplication now,
> 
> Is there?

Yes. There's the finished page (which I initially modified) and there's the script that creates that page (which I modify in this PR).

Reply:
> the finished page

That's like the Jamulus executable.  That's why it shouldn't be held under source control - it's _not_ source material, it should be output from the build process.

We just need to ensure the build process builds and publishes it correctly.

Reply:
I understand, and even agree, in principle. In reality, owning the "[care and feeding](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md)" of a PR is unrealistic when the implications become so far-reaching.

Reply:
Agreed - if the existing process _already_ works this way, it's not for this PR to change it.  It's just sad it does.  It needs an issue raised, at least, before the PR can be signed off, though, so the existing issue is tracked.

Reply:
https://github.com/jamulussoftware/jamulus/issues/3108